source/crud/tests/unified/create-null-ids.yml

```diff
@@ -0,0 +1,169 @@
+      id: &database database
+      client: client
+      databaseName: crud_id
+  - collection:
+      id: &collection collection
+      database: *database
+      collectionName: type_tests
+
+initialData:
+  - collectionName: type_tests
+    databaseName: crud_id
+    documents: []
+
+tests:
+
+  - description: inserting _id with type null via insertOne
+    operations:
+      - name: insertOne
+        object: *collection
+        arguments: {document: &null_id {_id: null}}
+      # We use countDocuments with a $type query to verify the insert of the correct BSON type
+      # this is to avoid client side type conversions (potentially common: undefined -> null)
+      - name: countDocuments
+        object: *collection
+        arguments: {filter: &null_id_filter {_id: {$type: 'null'}}}
+        expectResult: 1
+
+  - description: inserting _id with type null via insertMany
+    operations:
+      - name: insertMany
+        object: *collection
+        arguments: {documents: [*null_id]}
+      - name: countDocuments
+        object: *collection
+        arguments: {filter: *null_id_filter}
+        expectResult: 1
+
+  - description: inserting _id with type null via updateOne
+    operations:
+      - name: updateOne
+        object: *collection
+        arguments: {filter: *null_id, update: {$unset: {a: ''}}, upsert: true}
+      - name: countDocuments
+        object: *collection
+        arguments: {filter: *null_id_filter}
+        expectResult: 1
+
+  - description: inserting _id with type null via updateMany
+    operations:
+      - name: updateMany
+        object: *collection
+        arguments: {filter: *null_id, update: {$unset: {a: ''}}, upsert: true}
+      - name: countDocuments
+        object: *collection
+        arguments: {filter: *null_id_filter}
+        expectResult: 1
+
+  - description: inserting _id with type null via replaceOne
+    operations:
+      - name: replaceOne
+        object: *collection
+        arguments: {filter: {}, replacement: *null_id, upsert: true}
+      - name: countDocuments
+        object: *collection
+        arguments: {filter: *null_id_filter}
+        expectResult: 1
+
+  - description: inserting _id with type null via bulkWrite
+    operations:
+      - name: bulkWrite
+        object: *collection
+        arguments: {requests: [{insertOne: {document: *null_id}}]}
+      - name: countDocuments
+        object: *collection
+        arguments: {filter: *null_id_filter}
+        expectResult: 1
+
+  - description: inserting _id with type null via clientBulkWrite
+    runOnRequirements:
+      - minServerVersion: '8.0'
+    operations:
+      - name: clientBulkWrite
+        object: client
+        arguments: {models: [{insertOne: {namespace: crud_id.type_tests, document: *null_id}}]}
+      - name: countDocuments
+        object: *collection
+        arguments: {filter: *null_id_filter}
+        expectResult: 1
+
+  - description: inserting _id with type undefined via insertOne
```

Comment:
Note: I'm removing these tests in https://github.com/mongodb/specifications/pull/1713 as the server prohibits undefined values for `_id`.